File: src/jvmMain/java/resaver/pex/VDataNone.java
package resaver.pex;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import java.io.IOException;
import java.nio.ByteBuffer;

import static resaver.pex.DataType.NONE;

/**
 * VData that stores nothing.
 */
public class VDataNone extends VData {

    VDataNone() {
    }

    @Override
    public void write(@NotNull ByteBuffer output) throws IOException {
        output.put((byte) this.getType().ordinal());
    }

    @Override
    public int calculateSize() {
        return 1;
    }

    @Override
    public DataType getType() {
        return NONE;
    }

    @NotNull
    @Override
    public String toString() {
        return "NONE";
    }

    @Override
    public int hashCode() {
        int hash = 7;
        hash = 83 * hash + VDataNone.class.hashCode();
        return hash;
    }

    @Override
    public boolean equals(@Nullable Object obj) {
        if (this == obj) {
            return true;
        } else if (obj == null) {
            return false;
        } else return getClass() == obj.getClass();
    }

}

File: src/jvmMain/kotlin/resaver/pex/VDataNone.kt
package resaver.pex

import PlatformByteBuffer
import java.io.IOException

/**
 * VData that stores nothing.
 */
class VDataNone internal constructor() : VData() {
    @Throws(IOException::class)
    override fun write(output: PlatformByteBuffer?) {
        output?.put(type.ordinal.toByte())
    }

    override fun calculateSize(): Int {
        return 1
    }

    override val type: DataType
        get() = DataType.NONE

    override fun toString(): String {
        return "NONE"
    }

    override fun hashCode(): Int {
        var hash = 7
        hash = 83 * hash + VDataNone::class.hashCode()
        return hash
    }

    override fun equals(other: Any?): Boolean {
        return when {
            this === other -> {
                true
            }
            other == null -> {
                false
            }
            else -> javaClass == other.javaClass
        }
    }
}